Restore malloc hooks for mach-defpager until we have mlockall ready.

Index: glibc-2.24/malloc/malloc.c
===================================================================
--- glibc-2.24.orig/malloc/malloc.c
+++ glibc-2.24/malloc/malloc.c
@@ -1852,8 +1852,6 @@ static void *memalign_hook_ini (size_t a
 
 #if HAVE_MALLOC_INIT_HOOK
 void weak_variable (*__malloc_initialize_hook) (void) = NULL;
-compat_symbol (libc, __malloc_initialize_hook,
-	       __malloc_initialize_hook, GLIBC_2_0);
 #endif
 
 void weak_variable (*__free_hook) (void *__ptr,
